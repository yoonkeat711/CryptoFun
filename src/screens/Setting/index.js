import React, {useState, useEffect} from 'react';
import {View, Switch, SafeAreaView, Text} from 'react-native';
import AsyncStorage from '@react-native-community/async-storage';
import storageKey from '../../constants/storageKey';

const Setting = () => {
  const [isLocked, setIsLocked] = useState();

  const onSwitchToggle = async (value) => {
    setIsLocked(value);

    //
    await AsyncStorage.setItem(
      storageKey.isPasswordLocked,
      JSON.stringify(!isLocked),
    );
  };

  const getStorageLocked = (() => {
    return new Promise((resolve, reject) => {
      AsyncStorage.getItem(storageKey.isPasswordLocked)
        .then((value) => resolve(JSON.parse(value)))
        .catch((error) => reject(error));
    });
  })();

  useEffect(() => {
    getStorageLocked.then((value) => setIsLocked(value));
  }, []);

  return (
    <SafeAreaView>
      <View
        style={{
          flexDirection: 'row',
          justifyContent: 'space-between',
          padding: 16,
        }}>
        <Text>{'Lock'}</Text>
        <Switch onValueChange={onSwitchToggle} value={isLocked} />
      </View>
    </SafeAreaView>
  );
};

export default Setting;
