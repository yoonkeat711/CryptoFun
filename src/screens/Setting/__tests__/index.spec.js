import Setting from '../index';
import React from 'react';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Setting />).toJSON();
  expect(tree).toMatchSnapshot();
})
