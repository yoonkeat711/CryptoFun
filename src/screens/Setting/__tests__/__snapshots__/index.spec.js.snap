// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      }
    }
  >
    <Text>
      Lock
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      value={false}
    />
  </View>
</RCTSafeAreaView>
`;
